### DISCUSSION TITLE
SQL4CDS Doesnt Fire

### DISCUSSION THREAD

**Discussion by RichINSS** [MarkMpn/Sql4Cds#232] [Q&A]
Hi,

I'm hoping for a few pointers as I cant get the SQL4CDS add in to work with SSMS V18.12.1. I have installed the newest 6.4 package and can see the toolbar and the settings in the SSMS options window so I assume the add-in has installed correctly and loaded. That said, no matter what I try, I cant get a simple update statement to run. I can select data from the dataverse fine but any attempt at running an update just brings back the standard "UPDATE statement is not supported" with no visible attempt at conversion. What have I done wrong?

**Comment by MarkMpn** [comment]
Can you take a look at #141 for instructions on generating a log file from SSMS, then search that log file for any references to `Sql4Cds` or `SQL 4 CDS` which might give any indication of the problem?

Can you also include a screenshot of the SSMS connection dialog so I can check if there's anything in there that might mean the plugin doesn't pick up the connection as a valid Dataverse connection?

**Comment by RichINSS** [comment]
Thank you for the suggestion. I have run SSMS in debug mode and found the following error under SQL4CDS

System.NullReferenceException: Object reference not set to an instance of an object. at MarkMpn.Sql4Cds.Engine.TableSizeCache..ctor(IOrganizationService org, IAttributeMetadataCache metadata) in D:\a\1\s\MarkMpn.Sql4Cds.Engine\TableSizeCache.cs:line 27 at MarkMpn.Sql4Cds.SSMS.CommandBase.GetDataSource() at MarkMpn.Sql4Cds.SSMS.DmlExecute.OnExecuteQuery(String Guid, Int32 ID, Object CustomIn, Object CustomOut, Boolean& CancelDefault

![connection](https://user-images.githubusercontent.com/99200040/185386050-2a814ed5-0740-45b7-9dbf-ac2fceed6458.jpg)
)

**Reply by MarkMpn** [reply]
Could you try using the "Azure Active Directory - Universal with MFA" authentication type please? I can't test the password option as all my instances require MFA.

**Reply by RichINSS** [reply]
I have now tried using Universal with MFA but I am still getting the same issue

**Reply by MarkMpn** [reply]
OK, it's possible something is going wrong with the authentication to Dataverse. The best way for me to see what's happening here would be if you can run [Fiddler](https://www.telerik.com/download/fiddler-b) with HTTPS decryption enabled while you open SSMS and try to run the query. If that's possible, can you then save the log of the requests it's making and send it to me at sql4cds@markcarrington.dev?
